kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/EngineServiceDiscovery.scala

```diff
@@ -52,3 +61,59 @@ class EngineServiceDiscovery private(
     super.stop()
   }
 }
+
+object EngineServiceDiscovery extends Logging {
+
+  val mapper: ObjectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
+
+  def getEngineByPolicy(
+      zkClient: CuratorFramework,
+      namespace: String,
+      providePolicy: ProvidePolicy): Option[(String, Int)] = {
+    providePolicy match {
+      case RANDOM =>
+        Random.shuffle(getServiceNodesInfo(zkClient, namespace, silent = true)) match {
```

Comment:
I have a question, please correct me if there is any problem.Here All engines under namespce are obtained. If there is one engine, the create phase is not entered, and the engine state is always one.

Reply:
Just wait a moment, try to understand your comment.

Reply:
@zwangsheng, you're right, had improved the logic.